torch_glow/src/PyTorchModelLoader.cpp

```diff
@@ -339,60 +495,83 @@ void PyTorchModelLoader::loadConstant(const torch::jit::Node *ptNode) {
+      [this](const torch::jit::Node *node) { return loadConstant(node); };
+
+  nodeLoaderMapping_[at::Symbol::fromQualString("aten::mul")] =
+      [this](const torch::jit::Node *node) { return loadMul(node); };
+  nodeLoaderMapping_[at::Symbol::fromQualString("aten::mul_")] =
+      [this](const torch::jit::Node *node) { return loadMul(node); };
+
+  nodeLoaderMapping_[at::Symbol::fromQualString("aten::div")] =
+      [this](const torch::jit::Node *node) { return loadDiv(node); };
+  nodeLoaderMapping_[at::Symbol::fromQualString("aten::div_")] =
+      [this](const torch::jit::Node *node) { return loadDiv(node); };
+
+  nodeLoaderMapping_[at::Symbol::fromQualString("aten::add")] =
+      [this](const torch::jit::Node *node) { return loadAdd(node); };
+  nodeLoaderMapping_[at::Symbol::fromQualString("aten::add_")] =
+      [this](const torch::jit::Node *node) { return loadAdd(node); };
+
+  nodeLoaderMapping_[at::Symbol::fromQualString("aten::sub")] =
+      [this](const torch::jit::Node *node) { return loadSub(node); };
+  nodeLoaderMapping_[at::Symbol::fromQualString("aten::sub_")] =
+      [this](const torch::jit::Node *node) { return loadSub(node); };
+
+  nodeLoaderMapping_[at::Symbol::fromQualString("aten::relu")] =
+      [this](const torch::jit::Node *node) { return loadRelu(node); };
+  nodeLoaderMapping_[at::Symbol::fromQualString("aten::relu_")] =
+      [this](const torch::jit::Node *node) { return loadRelu(node); };
+
+  nodeLoaderMapping_[at::Symbol::fromQualString("aten::_convolution")] =
+      [this](const torch::jit::Node *node) { return loadConvolution(node); };
+
+  nodeLoaderMapping_[at::Symbol::fromQualString("aten::batch_norm")] =
+      [this](const torch::jit::Node *node) { return loadBatchNorm(node); };
+
+  nodeLoaderMapping_[at::Symbol::fromQualString("aten::max_pool2d")] =
+      [this](const torch::jit::Node *node) { return loadMaxPool2d(node); };
+}
+
+void PyTorchModelLoader::loadNode(const torch::jit::Node *node) {
+  auto kind = node->kind();
+  assert(nodeLoaderMapping_.count(kind));
+  return nodeLoaderMapping_.at(kind)(node);
 }
 
+/// knownSymbols contains the list of jit Symbols that are known to
+/// PyTorchModelLoader. This is populated the first time \ref
+/// PyTorchModelLoader::isNodeSupported is called and referenced on subsequent
+/// calls.
+static std::unordered_set<torch::jit::Symbol> knownSymbols;
+
 // static
 bool PyTorchModelLoader::isNodeSupported(const torch::jit::Node *node) {
-  // NOTE: Even though importing at::prim::Constant nodes is supported, don't
-  // add it here so that only the Constant nodes that are strictly necessary
-  // for values in the subgraph will be imported.
-  switch (node->kind()) {
-  case at::aten::mul:
-  case at::aten::div:
-  case at::aten::add:
-  case at::aten::sub:
-  case at::aten::relu:
-  case at::aten::_convolution:
-    return true;
-  default:
-    return false;
-  }
-  return false;
+  static std::once_flag flag;
+  std::call_once(flag, [&]() {
```

Comment:
this chunk of fancy code has already been put in "Rui's note: how to write graceful C++"

Reply:
> graceful C++
If it exists, I don't think this is it 😛